File: android/src/main/kotlin/com/kasem/media_picker_builder/MediaFile.kt
package com.kasem.media_picker_builder

import org.json.JSONObject

class MediaFile(
        var id: Long,
        var albumId: Long,
        var albumName: String,
        var dateAdded: Long,
        var path: String,
        var thumbnailPath: String?,
        /**
         * The orientation for the image expressed as degrees.
         * Only degrees 0, 90, 180, 270 will work.
         * */
        var orientation: Int,
        var mimeType: String?,

        /**
         * Video duration in milliseconds
         */
        var duration: Long?,
        var type: MediaType,
        var height: Int,
        var width: Int) {

    fun toJSONObject(): JSONObject {
        return JSONObject()
                .put("id", id.toString())
                .put("dateAdded", dateAdded)
                .put("path", path)
                .put("thumbnailPath", thumbnailPath)
                .put("orientation", orientation)
                .put("mimeType", mimeType)
                .put("duration", duration)
                .put("type", type.ordinal)
                .put("height", height)
                .put("width", width)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MediaFile

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }


    enum class MediaType {
        IMAGE, VIDEO
    }


}

File: ios/Classes/MediaFile.swift
//
//  MediaFile.swift
//  file_picker
//
//  Created by Kasem Mohamed on 6/29/19.
//

import Foundation

struct MediaFile : Codable {
    var id: String
    var dateAdded: Int? // seconds since 1970
    var path: String?
    var thumbnailPath: String?
    var orientation: Int
    var duration: Double?
    var mimeType: String?
    var type: MediaType
    var height: Int?
    var width: Int?
    
    init(id: String, dateAdded: Int?, path: String?, thumbnailPath: String?, orientation: Int, duration: Double?, mimeType: String?, type: MediaType, height: Int, width: Int) {
        self.id = id
        self.dateAdded = dateAdded
        self.path = path
        self.thumbnailPath = thumbnailPath
        self.orientation = orientation
        self.duration = duration
        self.mimeType = mimeType
        self.type = type
        self.height = height
        self.width = width
    }
}

enum MediaType: Int, Codable {
    case IMAGE
    case VIDEO
}


